client/src/main/fbthrift/com/facebook/thrift/transport/THttp2Client.java

```diff
@@ -0,0 +1,160 @@
+/* Copyright (c) 2023 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+package com.facebook.thrift.transport;
+
+import static java.time.LocalTime.now;
+import java.io.ByteArrayOutputStream;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Protocol;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.checkerframework.checker.units.qual.A;
+
+public class THttp2Client extends TTransport {
+
+    private final ByteArrayOutputStream requestBuffer = new ByteArrayOutputStream();
+    private ResponseBody responseBody = null;
+    private Map<String, String> customHeaders = null;
+    private static final Map<String, String> defaultHeaders = getDefaultHeaders();
+
+    private OkHttpClient client;
+    private final SSLSocketFactory sslFactory;
+
+    private final TrustManager trustManager;
+    private final String url;
+    private int connectTimeout = 0;
+    private int readTimeout = 0;
+
+
+    public THttp2Client(String url) throws TTransportException {
+        this(url, null, null);
+    }
+
+    public THttp2Client(String url, SSLSocketFactory sslFactory, TrustManager trustManager) throws TTransportException {
+        this.url = url;
+        this.sslFactory = sslFactory;
+        this.trustManager = trustManager;
+    }
+
+    public THttp2Client setConnectTimeout(int timeout) {
+        connectTimeout = timeout;
+        return this;
+    }
+
+    public THttp2Client setReadTimeout(int timeout) {
+        readTimeout = timeout;
+        return this;
+    }
+
+    public THttp2Client setCustomHeaders(Map<String, String> headers) {
+        customHeaders = headers;
+        return this;
+    }
+
+    public THttp2Client setCustomHeader(String key, String value) {
+        if (customHeaders == null) {
+            customHeaders = new HashMap<>();
+        }
+        customHeaders.put(key, value);
+        return this;
+    }
+
+    public void open() {
+        client = OkHttp3Util.getClient(connectTimeout, readTimeout, sslFactory, trustManager);
+    }
+
+    public void close() {
+        try {
+            if (responseBody != null) {
+                responseBody.close();
+                responseBody = null;
+            }
+
+            requestBuffer.close();
+        } catch (IOException e) {
+            // ignore
```

Comment:
swallow exception looks dangerous, better print some warning log?

Reply:
ok, I'll add the warning log.